besu/src/main/java/org/hyperledger/besu/controller/ConsensusScheduleBesuControllerBuilder.java

```diff
@@ -114,15 +119,29 @@ protected MiningCoordinator createMiningCoordinator(
       final MiningParameters miningParameters,
       final SyncState syncState,
       final EthProtocolManager ethProtocolManager) {
-    return besuControllerBuilderSchedule
-        .get(0L)
-        .createMiningCoordinator(
-            protocolSchedule,
-            protocolContext,
-            transactionPool,
-            miningParameters,
-            syncState,
-            ethProtocolManager);
+
+    final List<ForkSpec<MiningCoordinator>> miningCoordinatorForkSpecs =
+        besuControllerBuilderSchedule.entrySet().stream()
+            .map(
+                e ->
+                    new ForkSpec<>(
+                        e.getKey(),
+                        e.getValue()
+                            .createMiningCoordinator(
+                                protocolSchedule,
+                                protocolContext,
+                                transactionPool,
+                                miningParameters,
+                                syncState,
+                                ethProtocolManager)))
+            .collect(Collectors.toList());
+    final BftForksSchedule<MiningCoordinator> miningCoordinatorSchedule =
+        new BftForksSchedule<>(
+            miningCoordinatorForkSpecs.get(0),
```

Comment:
Is there any guarantee that the first miningCoordinatorForkSpec is for the genesis block?

Reply:
I tried and failed to expose a bug by ensuring miningCoordinatorForkSpecs.get(0) was not the genesis block.

The reason there is no bug is because the implementation of ForksSchedule (named BftForksSchedule here) renders the genesis fork parameter redundant.

```java
public ForksSchedule(final ForkSpec<C> genesisFork, final Collection<ForkSpec<C>> forks) {
    this.forks.add(genesisFork);
    this.forks.addAll(forks);
}
```
could be rewritten:
```java
public ForksSchedule(final Collection<ForkSpec<C>> forksIncludingGenesisInAnyOrder) {
    this.forks.addAll(forksIncludingGenesisInAnyOrder);
}
```

because of this...
```java
private final NavigableSet<ForkSpec<C>> forks =
      new TreeSet<>(
          Comparator.comparing((Function<ForkSpec<C>, Long>) ForkSpec::getBlock).reversed());
 ```

Reply:
Obviously the TreeSet with comparator is an implementation detail which shouldn't necessarily inform the API, but there's nothing else the genesisFork parameter is used for in this case. It would also simplify some other (especially test) code if we collapsed this into one parameter...in a separate PR of course :)

Reply:
Yep I can see it doesn't matter after all. Not sure why had seperate args for genesis and the other specs now but probably makes sense to collapse that in a seperate PR.